.github/workflows/call_test_package.yml

```diff
@@ -0,0 +1,142 @@
+        python -m pip install .[test]
+    # Run the tests with coverage and report add this end
+    # option for the coverage: --cov=. --cov-report=xml:... 
+    # option for mardown report: --md-report --md-report-output= ...
+    # option for html report: --self-contained-html --html=report-...
+    # option for duration report: --pytest-durations= number of second --pytest-resultlog=log-...
+    - name: Generate coverage report
+      run: |
+        python -m pytest\
+         --cov=. --cov-report=xml:cov/cov-${{env.REPORT_NAME}}.xml\
+         --md-report --md-report-output=test-results-${{env.REPORT_NAME}}.md\
+         --self-contained-html --html=report-${{env.REPORT_NAME}}.html\
+         --pytest-durations=10 --pytest-resultlog=log-${{env.REPORT_NAME}}.txt
+    ## apploading results
+    - name: Upload HTML test result
+      if: ${{ always() && !cancelled()}}
+      id: html_report
+      uses: actions/upload-artifact@v4
+      with:
+        name: pytest-results-html-${{env.REPORT_NAME}}
+        path: report-${{env.REPORT_NAME}}.html
+        if-no-files-found: error
+    # add the link for html report to the mardown file
+    - name: Save link HTML in file
+      if: ${{ always() && !cancelled()}}
+      run: 
+        echo -e "\n [HTML summary]("${{ steps.html_report.outputs.artifact-url }}")" >> test-results-${{env.REPORT_NAME}}.md
+    - name: Upload md test result 
+      if: ${{ always() && !cancelled()}}
+      uses: actions/upload-artifact@v4
+      with:
+        name: pytest-results-md-${{env.REPORT_NAME}}
+        path: test-results-${{env.REPORT_NAME}}.md 
+        if-no-files-found: error
+    - name: Upload duration test result
+      if: ${{ always() && !cancelled()}}
+      uses: actions/upload-artifact@v4
+      with:
+        name: pytest-results-log-${{env.REPORT_NAME}}
+        path: log-${{env.REPORT_NAME}}.txt
+        if-no-files-found: error
+    - name: Upload coverage data
+      if: ${{ always() && !cancelled()}}
+      uses: actions/upload-artifact@v4
+      with:
+        name: cov-results-${{env.REPORT_NAME}}
+        path: cov/cov-${{env.REPORT_NAME}}.xml
+        if-no-files-found: error
+
+  merge_md_test_data:
+    # for more details
+    # https://github.com/thombashi/pytest-md-report
+    needs: build-test
+    runs-on: ubuntu-latest
+    if: ${{ failure() }}
+    steps:
+    # get report from previous job 
+    - name: Get data
+      uses: actions/download-artifact@v4
+      with:
+        pattern: pytest-results-md-*-os-*
+    # create a comment in the PR with the failling tests
+    # for more details: https://github.com/marocchino/sticky-pull-request-comment
+    - name: Render the report to the PR when tests fail
+      uses: marocchino/sticky-pull-request-comment@v2
+      with:
+        header: test-report
+        recreate: true
+        path: ./pytest-results-md-3.13-os-macos-latest/test-results-3.13-os-macos-latest.md     #CHECK: need to be updated when new version are available
```

Comment:
yeah you may want to have env variable to keep track of which OS and python version you want to use as 'source' to generate the report

Reply:
In this case, I chose to summit only one report because it's the first to run, i.e. the first to fail, and to avoid having too many notifications.

Reply:
I finally put all the reports on circleci.